blockchains/eth/eth_worker.js

```diff
@@ -0,0 +1,254 @@
+const Web3 = require('web3')
+const jayson = require('jayson/promise');
+const { filterErrors } = require('blockchain-utils/eth')
+const constants = require('./lib/constants')
+const { logger } = require('../../lib/logger')
+const { injectDAOHackTransfers, DAO_HACK_FORK_BLOCK } = require('./lib/dao_hack')
+const { addGenesisTransfers } = require('./lib/genesis_transfers')
+const { transactionOrder, stableSort } = require('./lib/util')
+const BaseWorker = require('../../lib/worker_base')
+
+
+class ETHWorker extends BaseWorker {
+  constructor() {
+    super()
+
+    logger.info(`Connecting to parity node ${constants.PARITY_NODE}`)
+    this.web3 = new Web3(new Web3.providers.HttpProvider(constants.PARITY_NODE))
+    this.parityClient = jayson.client.http(constants.PARITY_NODE);
+  }
+
+  fetchEthInternalTrx(fromBlock, toBlock) {
+    return this.parityClient.request('trace_filter', [{
+      fromBlock: this.web3.utils.numberToHex(fromBlock),
+      toBlock: this.web3.utils.numberToHex(toBlock)
+    }]).then((data) => {
+      const traces = filterErrors(data["result"])
+
+      return traces
+        .filter((trace) =>
+          trace["action"]["value"] != "0x0" &&
+          trace["action"]["balance"] != "0x0" &&
+          !(trace["type"] == "call" && trace["action"]["callType"] != "call")
+        )
+    })
+  }
+
+  decodeTransferTrace(trace, blocks) {
+    const web3 = this.web3
+    const timestamp = web3.utils.hexToNumber(blocks.get(trace["blockNumber"]).timestamp)
+
+    // Block & uncle rewards
+    if (trace["type"] == "reward") {
+      return {
+        from: `mining_${trace["action"]["rewardType"]}`,
+        to: trace["action"]["author"],
+        value: parseFloat(web3.utils.hexToNumberString(trace["action"]["value"])),
+        valueExactBase36: web3.utils.toBN(trace["action"]["value"]).toString(36),
```

Comment:
Very similar exporters like ERC20 and ETH would share a lot of code, we can move that to a common utils file. In the common case however there would be no reuse of code under different /blockchains/* directories.

Reply:
The DRY principle states: 
> Every piece of knowledge must have a single, unambiguous, authoritative representation within a system

I don't try to follow it blindly however if something is repeated three times it fires an alarm in my head and try to make it reusable. 
In the current situation that line is repeated four times: `web3.utils.toBN(trace["action"]["value"]).toString(36)`

Extracting to a function in the same file: 
```python 
def parse_value_exact_base_36(field):
  return web3.utils.toBN(field).toString(36)
end
```
will have the following benefits:
1) Will be more readable. When I am scanning a file and start to feel that lines are similar I try to find the differences  between them. 
Here the difference is clear(the field):
 ```python
valueExactBase36: parse_value_exact_base_36(trace["action"]["value"]) 
```
2) It will be a lot easier to change
3) It will be a lot easier to write unit test for that function. In this case it will be very handy to write such test - it will document the input and the output of the function.

Reply:
OK i'm sold. Once I started doing it - it turned out that there are only four unique types of invocations. Now moved into four helper methods.